docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc

```diff
@@ -0,0 +1,82 @@
+
+[[painless-bucket-selector-agg-context]]
+=== Bucket Selector aggregation context
+
+Use a Painless script in an
+{ref}/search-aggregations-pipeline-bucket-selector-aggregation.html[`bucket_selector` aggregation]
+to determine if a bucket should be retained or filtered out.
+
+==== Variables
+
+`params` (`Map`, read-only)::
+        User-defined parameters passed in as part of the query. The parameters
+        include values defined as part of the `buckets_path`.
+
+==== Return
+
+boolean::
+        True if the the bucket should be retained, false if the bucket should be filtered out.
+
+==== API
+
+
+To run this example, first follow the steps in <<painless-context-examples, context examples>>.
+
+The painless context in a `bucket_selector` aggregation provides a `params` map.  This map contains both
+user-specified custom values, as well as the values from other aggregations specified in the `buckets_path`
+property.
+
+For our example, we are going to find the `max` of each bucket, add in a user-defined "base" value, then keep all
+the buckets that are greater than `10`.
+
```

Comment:
Same as for bucket script:

This example finds the `max` of each bucket, adds a user-specified base_cost, and retains all of the buckets that are greater than `10`.